docs/app-dev/indexing-transactions.md

```diff
@@ -4,9 +4,14 @@ order: 6
 
 # Indexing Transactions
 
-Tendermint allows you to index transactions and later query or subscribe to their results.
-
-Events can be used to index transactions and blocks according to what happened
+Tendermint allows you to index transactions and blocks and later query or
+subscribe to their results. Both transactions and blocks are indexed by their
+respective events. However, transactions are also indexed by a primary key which
+includes the transaction hash and maps to and stores the corresponding `TxResult`.
+Blocks are indexed by a primary key which includes the block height and maps to
+and stores the block height, i.e. the block itself is never stored.
+
+Events are be used to index transactions and blocks according to what happened
 during their execution. Note that the set of events returned for a block from
```

Comment:
> Note that the set of events returned for a block from
`BeginBlock` and `EndBlock` are merged. In case both methods return the same
type, only the key-value pairs defined in `EndBlock` are used.

@melekes is this true? Before this PR, blocks were not even indexed, so what is this even referring to?